banks/views.py

```diff
@@ -77,3 +85,63 @@ def post(self, request):
             new_account.save()
             return redirect('user-account-list')
         return render(request, 'create_account.html', {'form': form})
+
+
+class BankListApiView(APIView):
+    def get(self, request):
+        banks = Bank.objects.all()
+        data = [{'id': bank.id, 'name': bank.name, 'is_islamic': bank.is_islamic} for bank in banks]
+        return Response(data)
+
+
+class BankViewSet(ViewSet):
+    def list(self, request):
+        banks = Bank.objects.all()
+        data = [{'id': bank.id, 'name': bank.name, 'is_islamic': bank.is_islamic} for bank in banks]
+        return Response(data)
+
+
+class BankListGenericView(ListAPIView):
+    queryset = Bank.objects.all()
+    serializer_class = BankSerializer
```

Comment:
study `def list()` method of ListAPIView

similarly def create() for CreateAPIView

Reply:
then study the RetrieveUpdateDestroyAPIView